src/state/opamRepositoryState.ml

```diff
@@ -111,6 +111,53 @@ let load_opams_from_dir repo_name repo_root =
     (fun () -> aux OpamPackage.Map.empty (OpamRepositoryPath.packages_dir repo_root))
     ~finally:OpamConsole.clear_status
 
+let load_opams_incremental repo_name repo_root diffs rt =
+  let existing_opams =
+    match OpamRepositoryName.Map.find_opt repo_name rt.repo_opams with
+    | Some opams -> opams
+    | None -> OpamPackage.Map.empty
+  in
+  let process_file acc file ~is_removal =
```

Comment:
I'm not sure if it's done elsewhere or something, but this function seems to assume every files that can change are opam files. What about changes from their `files` directory or other non-opam file related changes?